google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardWebIntegration.java

```diff
@@ -0,0 +1,128 @@
+ */
+
+package com.google.cloud.tools.intellij.appengine.facet;
+
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineCloudType;
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineServerConfiguration;
+import com.google.cloud.tools.intellij.debugger.CloudDebugConfigType;
+import com.google.cloud.tools.intellij.debugger.CloudDebugRunConfiguration;
+
+import com.intellij.execution.RunManager;
+import com.intellij.execution.RunnerAndConfigurationSettings;
+import com.intellij.execution.configuration.ConfigurationFactoryEx;
+import com.intellij.execution.configurations.ConfigurationFactory;
+import com.intellij.execution.configurations.ModuleRunConfiguration;
+import com.intellij.framework.addSupport.FrameworkSupportInModuleProvider;
+import com.intellij.ide.util.frameworkSupport.FrameworkSupportModel;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.ModifiableRootModel;
+import com.intellij.openapi.roots.libraries.Library;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.packaging.artifacts.Artifact;
+import com.intellij.packaging.artifacts.ArtifactType;
+import com.intellij.remoteServer.ServerType;
+import com.intellij.remoteServer.configuration.RemoteServer;
+import com.intellij.remoteServer.configuration.RemoteServersManager;
+import com.intellij.remoteServer.impl.configuration.deployment.DeployToServerConfigurationType;
+import com.intellij.remoteServer.impl.configuration.deployment.DeployToServerConfigurationTypesRegistrar;
+import com.intellij.remoteServer.impl.configuration.deployment.DeployToServerRunConfiguration;
+import com.intellij.util.containers.ContainerUtil;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+/**
+ * @author nik
+ */
+public abstract class AppEngineStandardWebIntegration {
+
+  public static AppEngineStandardWebIntegration getInstance() {
+    return ServiceManager.getService(AppEngineStandardWebIntegration.class);
+  }
+
+  @Nullable
+  public abstract VirtualFile suggestParentDirectoryForAppEngineWebXml(@NotNull Module module,
+      @NotNull ModifiableRootModel rootModel);
+
+  @NotNull
+  public List<ArtifactType> getAppEngineTargetArtifactTypes() {
+    return ContainerUtil
+        .packNullables(getAppEngineWebArtifactType(), getAppEngineApplicationArtifactType());
+  }
+
+  @NotNull
+  public abstract ArtifactType getAppEngineWebArtifactType();
+
+  @Nullable
+  public abstract ArtifactType getAppEngineApplicationArtifactType();
+
+  @NotNull
+  public abstract List<FrameworkSupportInModuleProvider.FrameworkDependency>
+      getAppEngineFrameworkDependencies();
+
+  public abstract void setupJpaSupport(@NotNull Module module, @NotNull VirtualFile persistenceXml);
+
+  public abstract void setupRunConfigurations(@Nullable Artifact artifact, @NotNull Project project,
```

Comment:
The remaining setup*RunConfiguration methods here could also be turned private and disappear from this class API.